libs/template/templates/default-python/template/{{.project_name}}/tests/main_test.py.tmpl

```diff
@@ -1,21 +1,25 @@
-from databricks.connect import DatabricksSession
-from pyspark.sql import SparkSession
-from {{.project_name}} import main
-
-# Create a new Databricks Connect session. If this fails,
-# check that you have configured Databricks Connect correctly.
-# See https://docs.databricks.com/dev-tools/databricks-connect.html.
 {{/*
   The below works around a problematic error message from Databricks Connect.
   The standard SparkSession is supported in all configurations (workspace, IDE,
   all runtime versions, CLI). But on the CLI it currently gives a confusing
-  error message if SPARK_REMOTE is not set. We can't directly use
-  DatabricksSession.builder in main.py, so we're re-assigning it here so
-  everything works out of the box, even for CLI users who don't set SPARK_REMOTE.
+  error message if SPARK_REMOTE is not set. Here we import the DatabricksSession
+  as SparkSession so the rest of the code looks as expected.
 */}}
-SparkSession.builder = DatabricksSession.builder
-SparkSession.builder.getOrCreate()
+from databricks.connect import DatabricksSession as SparkSession
+from pytest import fixture
+from {{.project_name}} import main
+
+
+@fixture(scope="session")
+def spark():
+    # Create a new Databricks Connect session. If this fails, you should
+    # check that you have configured Databricks Connect correctly.
+    # See https://docs.databricks.com/dev-tools/databricks-connect.html.
+    spark = SparkSession.builder.getOrCreate()
+    yield spark
+    spark.stop()
```

Comment:
Are those two lines required? I've noticed that our docs [here](https://learn.microsoft.com/en-us/azure/databricks/dev-tools/vscode-ext/dev-tasks/pytest#step-1-create-the-tests) simply return the spark from the fixture

Reply:
I don't think we need them. I'll take them out

Reply:
What does it mean to stop a SC session?

Reply:
Fixtures for pytest belong in `conftest.py`, also see [docs](https://docs.pytest.org/en/7.1.x/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files).

Reply:
Thanks, I didn't know about `conftest.py`. We'll probably go with a solution that doesn't need a fixture